streams/src/main/java/org/apache/kafka/streams/KafkaStreams.java

```diff
@@ -95,6 +97,7 @@
     private static final String JMX_PREFIX = "kafka.streams";
     public static final int DEFAULT_CLOSE_TIMEOUT = 0;
     private final StreamThread[] threads;
+    private final Map<Long, StreamThread.State> threadState;
```

Comment:
Probably want to make this a `ConcurrentHashMap` or `synchronize` access to it. It can be modified etc from multiple threads